internal/controller/streams.go

```diff
@@ -197,105 +164,225 @@ func (sm *streamManager) BootUpStreamAndReconnect(ctx context.Context, cancel co
 		return err
 	}
 	snapshotCompleted.Done()
+
+	<-ctx.Done()
+	return ctx.Err()
+}
+
+// StreamLogs handles the log stream.
+func (sm *streamManager) StreamLogs(ctx context.Context, cancel context.CancelFunc) error {
+	errCh := make(chan error, 1)
+
+	go func() {
+		errCh <- sm.bufferedGrpcSyncer.ListenToLogStream()
+	}()
+
+	select {
+	case <-ctx.Done():
+		cancel()
```

Comment:
This is unnecessary, because the context gets canceled anyway on every return by the caller, cf. https://github.com/illumio/cloud-operator/pull/51/files#diff-65d4066e2022c168a97a2083474d348fa6d5d1fffb6e8d7378282935ad187940R261
```suggestion
```